pkg/util/trivy/trivy.go

```diff
@@ -47,52 +48,64 @@ const (
 	LicenseAnalyzers    = "license"
 )
 
+// ContainerdAccessor is a function that should return a containerd client
+type ContainerdAccessor func() (cutil.ContainerdItf, error)
+
 // CollectorConfig allows to pass configuration
 type CollectorConfig struct {
-	ArtifactOption     artifact.Option
-	CacheProvider      CacheProvider
-	ClearCacheOnClose  bool
-	ContainerdAccessor func() (containerdUtil.ContainerdItf, error)
-	CheckDiskUsage     bool
-	MinAvailableDisk   uint64
+	CacheProvider     CacheProvider
+	ClearCacheOnClose bool
 }
 
 // Collector uses trivy to generate a SBOM
-type collector struct {
+type Collector struct {
 	config     CollectorConfig
 	cache      cache.Cache
 	applier    local.Applier
 	detector   local.OspkgDetector
 	dbConfig   db.Config
 	vulnClient vulnerability.Client
 	marshaler  *cyclonedx.Marshaler
-	disk       filesystem.Disk
+}
+
+var globalCollector *Collector
+
+func getDefaultArtifactOption(enabledAnalyzers []string, root string) artifact.Option {
+	option := artifact.Option{
+		Offline:           true,
+		NoProgress:        true,
+		DisabledAnalyzers: DefaultDisabledCollectors(enabledAnalyzers),
+		Slow:              true,
+		SBOMSources:       []string{},
+		DisabledHandlers:  DefaultDisabledHandlers(),
+	}
+
+	if len(enabledAnalyzers) == 1 && enabledAnalyzers[0] == OSAnalyzers {
+		option.OnlyDirs = []string{"etc", "var/lib/dpkg", "var/lib/rpm", "lib/apk"}
+		if root != "" {
+			option.Slow = false
```

Comment:
Why not keeping `Slow` for FS scan?

Reply:
Not sure why we should use Slow neither :-) Anyway, I changed it to `true`